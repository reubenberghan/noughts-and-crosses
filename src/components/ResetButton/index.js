import React from 'react'
import styled from 'styled-components'

const StyledResetButton = styled.button`
  cursor: pointer;
  height: 5vh;
  justify-self: center;
  width: 20vh;
`

StyledResetButton.displayName = 'StyledResetButton'

export default function ResetButton ({ handleClick }) {
  return <StyledResetButton onClick={handleClick}>Reset Game</StyledResetButton>
}
