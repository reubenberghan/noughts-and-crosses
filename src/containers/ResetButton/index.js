import { connect } from 'react-redux'

import ResetButton from '../../components/ResetButton'
import { resetGame } from '../../state'

function mapDispatchToProps (dispatch) {
  return {
    handleClick: () => dispatch(resetGame())
  }
}

export default connect(undefined, mapDispatchToProps)(ResetButton)
