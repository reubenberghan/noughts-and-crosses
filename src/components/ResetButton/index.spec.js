import React from 'react'
import { shallow } from 'enzyme'

import ResetButton from '.'

describe('components:ResetButton', () => {
  it('renders the ResetButton component with proper styles', () => {
    expect(toJson(shallow(<ResetButton />).dive())).toMatchSnapshot()
  })
})
