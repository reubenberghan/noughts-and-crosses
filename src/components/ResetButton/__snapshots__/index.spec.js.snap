// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:ResetButton renders the ResetButton component with proper styles 1`] = `
.c0 {
  cursor: pointer;
  height: 5vh;
  justify-self: center;
  width: 20vh;
}

<button
  className="c0"
>
  Reset Game
</button>
`;
